CMake/SourceGroupByFolder.cmake

```diff
@@ -0,0 +1,33 @@
+# Adapted from https://github.com/mbrucher/AudioTK/blob/master/CMake/Utilities.cmake
+# License: https://github.com/mbrucher/AudioTK/blob/master/LICENSE.md
+
+macro(source_group_by_folder target)
```

Comment:
You may want to consider using: https://github.com/MerryMage/dynarmic/blob/master/CMakeModules/CreateDirectoryGroups.cmake which is both smaller, and not a macro, so it won't pollute the namespace around it with variables.

Reply:
Tested it and I like it a lot. This function does not deal with header files though. Any suggestions what the best way to do so would be here?

Reply:
This is just because the current CMake targets in Dolphin currently don't specify the headers in their `add_library()` or `add_executable()` invocation (we *should* be specifying them). Once they're added to the respective invocations, then they'll show up just fine, since the above linked function operates on the source files provided by the target itself (which is nice, because it means you don't traverse the whole file tree repeatedly)

This is mostly a historical thing (which this change addresses essentially), given Windows was the only "IDE-centric" build environment, and it was using its own project files (on *nix, you usually just generate Makefiles, or Ninja build files, and run it on the CLI, so visual organization wasn't really a thing).